spec/lib/ai/agent_spec.rb

```diff
@@ -41,13 +42,53 @@
           runtime_context: {
           },
           max_retries: 5,
-          max_steps: 10
+          max_steps: 10,
+          telemetry: anything
         }
       ).and_call_original
 
       agent.generate_text(messages: [Ai.user_message('Hello')], max_retries: 5, max_steps: 10)
     end
 
+    it 'passes telemetry settings to client' do
+      telemetry_settings = Ai::TelemetrySettings.new(
+        is_enabled: true,
+        record_inputs: true,
+        record_outputs: true,
+        function_id: 'test-function',
+        metadata: { 'agent.name' => 'test-agent' }
+      )
+
+      expect(client).to receive(:generate).with(
+        'test',
+        messages: anything,
+        options: {
+          runtime_context: {},
+          max_retries: 2,
+          max_steps: 5,
+          telemetry: telemetry_settings
+        }
+      ).and_call_original
+
+      agent.generate_text(messages: [Ai.user_message('Hello')], telemetry: telemetry_settings)
+    end
+
+    it 'uses default telemetry settings when none provided' do
```

Comment:
Use `context` to organize specs with conditions (my rule of thumb is that when I have an `if` or `when`  in a `it` it means I need a context).